[cli] Readme: formatting, updated E2E instructions (#34633)
expo/expo: packages/@expo/cli/README.md

npx expo
```

> > ⭐️ Be sure to star the Expo GitHub repo if you enjoy using the project!

Reviewer: Can we keep this for now? Callouts do help 😄

This CLI has the following purposes:

- Be a minimal interface for starting a local development server that emulates a production EAS Updates server. The development server is the proxy between a native runtime (Expo Go, Dev Client) and a JS Bundler (Metro, Webpack).

>   - To accomplish secure manifest signing (think https/TSL/SSL for web (required for sandboxing AsyncStorage, Permissions, etc.) we need an authenticated Expo user account. This is the only reason we include the authentication commands `login`, `logout`, `whoami`, `register`. Standard web CLIs don't have authentication commands because they either don't set up https or they use emulation via packages like `devcert`.

Reviewer: The double `))` is correct here, the parentheses are opened twice.
> ... (think https/TSL/SSL for web (required for ...

Or we need to reword into something like:

```suggestion
  - To accomplish secure manifest signing for the web, like AsyncStorage or Permissions where both require https/SSL, we need an authenticated Expo account. This is the only reason we include the authentication commands `login`, `logout`, `whoami`, `register`. Standard web CLIs don't have authentication commands because they either don't set up https or they use emulation via packages like `devcert`.
```

Author: Good catch, I like the alternate wording! 👌

### E2E Testing Guidelines

- E2E tests should be resilient and reliable, be sure to give them plenty of time for network requests.

> - When testing locally you had to have Bun installed, it can be done via Volta - `volta install bun`.

Reviewer: Let's link to Bun's docs on how to install instead. I'm sure most people aren't using Volta, and the docs include all kinds of different installation methods (plus we don't need to keep this up-to-date) - https://bun.sh/docs/installation

- When testing locally you should attempt to reuse node modules for faster results. In the `npx expo prebuild` and `npx expo start` commands for instance, we utilize a helper method that will default to reusing a project + **node_modules** when run locally. This can be [toggled off](https://github.com/expo/expo/blob/11a5a4d27b7e1c8e4d6ddf0401397d789d89f52a/packages/%40expo/cli/e2e/__tests__/utils.ts#L174) to bootstrap a fresh project every time.
- When bootstrapping test projects, utilize the temporary folder `os.tmpdir()` as this folder is automatically cleaned up when the computer restarts.

## Coming from Expo CLI

> > [!important]

Reviewer: I don't think it's that important tbh 😄 It's great for context, no one should be on `expo-cli` anymore.
